### PR TITLE
Improved `test_storage_s3_get_unstable`

### PR DESCRIPTION
**Pull request by excitoon**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Progress on #23911

### CODE REVIEW

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: tests/integration/test_storage_s3/s3_mocks/unstable_server.py [hunk 1]
```diff
@@ -11,7 +11,11 @@ def gen_n_digit_number(n):
     return random.randint(10**(n-1), 10**n-1)
 
 
+sum_4 = 0
```
Comment (alexey-milovidov) [inline tests/integration/test_storage_s3/s3_mocks/unstable_server.py:14]: Mysterious variable name.

File: tests/integration/test_storage_s3/s3_mocks/unstable_server.py [hunk 2]
```diff
@@ -24,13 +28,14 @@ def add_number():
...
         add_number()
+    sum_4 += row[-1]
 
     line = ",".join(map(str, row)) + "\n"
     return line.encode()
 
 
 random.seed("Unstable server/1.0")
-lines = b"".join((gen_line() for _ in range(500000)))
+lines = b"".join((gen_line() for _ in range(500000))) + f"0,0,0,{-sum_4}\n".encode()
```
Comment (alexey-milovidov) [inline tests/integration/test_storage_s3/s3_mocks/unstable_server.py:38]: What's the logic behind this?
Please write in comments in the code.

**Comment by excitoon** [comment]
@alexey-milovidov This patch improves testing capabilities of the test, but I was unable to reproduce #23911 on it anyway. I guess the problem is only disk-related.

**Comment by alexey-milovidov** [comment]
@mergifyio update

**Comment by alexey-milovidov** [comment]
@excitoon You did not provide access to your fork.

**Comment by excitoon** [comment]
@alexey-milovidov Rebased it.

**Comment by alexey-milovidov** [comment]
Cool!

**Pull request merged**